source/extensions/filters/http/ratelimit/ratelimit_headers.cc

```diff
@@ -0,0 +1,91 @@
+#include "extensions/filters/http/ratelimit/ratelimit_headers.h"
+
+#include "common/http/header_map_impl.h"
+
+#include "absl/strings/str_cat.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace HttpFilters {
+namespace RateLimitFilter {
+
+Http::RegisterCustomInlineHeader<Http::CustomInlineHeaderRegistry::Type::ResponseHeaders>
+    x_rate_limit_limit_handle(Http::CustomHeaders::get().XRateLimitLimit);
+Http::RegisterCustomInlineHeader<Http::CustomInlineHeaderRegistry::Type::ResponseHeaders>
+    x_rate_limit_remaining_handle(Http::CustomHeaders::get().XRateLimitRemaining);
+Http::RegisterCustomInlineHeader<Http::CustomInlineHeaderRegistry::Type::ResponseHeaders>
+    x_rate_limit_reset_handle(Http::CustomHeaders::get().XRateLimitReset);
```

Comment:
I don't think these are needed as you are just adding directly by reference below which seems fine to me.